File: post-novel-web/src/main/java/com/postnovel/web/postnovelweb/endpoint/PostNovelEndpoint.java
package com.postnovel.web.postnovelweb.endpoint;

import com.postnovel.web.postnovelweb.domain.Post;
import com.postnovel.web.postnovelweb.client.PostNovelService;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RestController;

import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;

@RestController
public class PostNovelEndpoint {

    private PostNovelService postNovelService;

    public PostNovelEndpoint(PostNovelService postNovelService) {
        this.postNovelService = postNovelService;
    }

    @GetMapping(value = "/posts/{id}", produces = APPLICATION_JSON_VALUE)
    public Post retrieveAPost(@PathVariable int id) {
        return postNovelService.getPostById(id);
    }
}

File: src/main/kotlin/com/wiremock/example/springbootwiremockexample/web/PostNovelEndpoint.kt
package com.wiremock.example.springbootwiremockexample.web

import com.wiremock.example.springbootwiremockexample.client.PostClient
import com.wiremock.example.springbootwiremockexample.domain.Post
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

@RestController
class PostNovelEndpoint(
        @Autowired
        private val postClient: PostClient
) {

    @GetMapping("/posts")
    fun retrieveAllPosts(): List<Post> {
        return postClient.getAllPosts()
    }
}